sources/packages/backend/apps/api/src/route-controllers/overaward/overaward.institution.controller.ts

```diff
@@ -0,0 +1,54 @@
+import { Controller, Param, ParseIntPipe, Get } from "@nestjs/common";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty } from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import {
+  OverawardBalanceAPIOutDTO,
+  OverawardDetailsAPIOutDTO,
+} from "./models/overaward.dto";
+import { OverawardControllerService } from "..";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("overaward")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-overaward`)
+export class OverawardInstitutionController extends BaseController {
+  constructor(
+    private readonly overawardControllerService: OverawardControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the overaward balance of a student.
+   * @param studentId student.
+   * @returns overaward balance for student.
+   */
+  @ApiNotFoundResponse({
```

Comment:
are we throwing this error in this controller?

Reply:
No. I have removed it.